### ISSUE TITLE
Image planner non stop cycling of targets

### ISSUE THREAD

**Issue by astromiester1** [rrskybox/ImagePlanner#14]
Image planner does no lock in on selected target from nightshirt latest build. Instead, IP cycles through all my 60 targets. I see you moved this function to  IP list... The behavior still happens in IP.... The list never stabilizes and cycles over and over again repeating all your targets.

**Comment by rrskybox** [comment]
60 targets is a pretty big list.  TSX must be sampled for each target to compile the Target List -- it's very slow.   I'm guessing that you might be hitting the List button (in frustration) multiple times while the first click is still processing, so it looks like its looping.  I modified that button so that it will update only once while in the middle of processing (I'd like to make it so that you can't reclick at all, but it doesn't seem to want to so far).  I put a new release (Build 107) up.  This work-around breaks the Prospect Form update operation, but I'll work on that later.

**Comment by astromiester1** [comment]
I made a mistake ..Its about 25...Sent via the Samsung Galaxy S22 5G, an AT&T 5G smartphone
-------- Original message --------From: Rick McAlister ***@***.***> Date: 5/17/24  12:20 PM  (GMT-05:00) To: rrskybox/ImagePlanner ***@***.***> Cc: astromiester1 ***@***.***>, Author ***@***.***> Subject: Re: [rrskybox/ImagePlanner] Image planner non stop cycling of targets (Issue #14) 
60 targets is a pretty big list.  TSX must be sampled for each target to compile the Target List -- it's very slow.   I'm guessing that you might be hitting the List button (in frustration) multiple times while the first click is still processing, so it looks like its looping.  I modified that button so that it will update only once while in the middle of processing (I'd like to make it so that you can't reclick at all, but it doesn't seem to want to so far).  I put a new release (Build 107) up.  This work-around breaks the Prospect Form update operation, but I'll work on that later.

**Comment by rrskybox** [comment]
I can't seem to replicate this behavior even with a big target list.  However, as a shot in the dark, I modified the Table List Form event handler (which also removed and extra and unintentional refresh).  That's the only way I can think of that the table refresh could be could be looping on a sort, but I'll keep looking at it.  Build 108.

**Comment by astromiester1** [comment]
Rick, if it helps.. Here is my May target list from google drive....

https://drive.google.com/drive/folders/1SaFl-VqCv_bUxS0dVYjTHj4XhswxcXmX?usp=sharing

**Comment by rrskybox** [comment]
Thanks for your target set.  However, I still don't see the behavior you describe.  I did widen the Target List window some to accommodate the longer target names you have there.  As of the last build or so, a single click on a column header shouldn't even invoke a refresh, just sort the table without even entering the application code.  (I just added another change that prevents double clicks on a column header to invoke a refresh as well -- Build 111).  I'll keep poking at it though.

**Comment by astromiester1** [comment]
What version of Nightshift(NS) should be used with IP version 111? I am running both in 64bit mode. Can I test IP without NS to test?

**Comment by astromiester1** [comment]
Testing IP 111 64 bit by itself, after targets populate list, if I sort on altitude then select a target then a different target, the target list continues to cycle over and over wo any updates to the graph.

**Comment by astromiester1** [comment]
If I shorten up the target list, every time I select a new target from the list it refreshs the target list and updates the new IP graph. I would think the list should not refresh unless instructed to do so. Instead it refreshes the list each time you select a target. As you have more targets the wait time is huge... I have 60 or so.

**Comment by astromiester1** [comment]
Where in TSX is StarChart Refresh Rate?

**Comment by rrskybox** [comment]
Edit -> Preferences -> Advanced -> Target Frame Rate

Set to 1 (then restart, I think)

**Comment by rrskybox** [comment]
ImagePlanner is modified to update the Target List form only when the date selection changes.  I also changed the grid background to a reddish color during refresh to indicate that a  refresh is underway (helpful in the case of large target lists).  Build 112.

Closing this issue accordingly.

**Comment by astromiester1** [comment]
Is nightshift(NS) now not required to work with IP in latest builds since target list moved to IP... if so, looks like NS just a GUI to operate a list of other tools..ThanksDaleSent via the Samsung Galaxy S22 5G, an AT&T 5G smartphone
-------- Original message --------From: Rick McAlister ***@***.***> Date: 5/18/24  8:20 PM  (GMT-05:00) To: rrskybox/ImagePlanner ***@***.***> Cc: astromiester1 ***@***.***>, Author ***@***.***> Subject: Re: [rrskybox/ImagePlanner] Image planner non stop cycling of targets (Issue #14) 
ImagePlanner is modified to update the Target List form only when the date selection changes.  I also changed the grid background to a reddish color during refresh to indicate that a  refresh is underway (helpful in the case of large target lists).  Build 112.
Closing this issue accordingly.

**Comment by astromiester1** [comment]
Works great Rick.....

**Comment by rrskybox** [comment]
ImagePlanner has always been standalone -- in fact, it was written well before NightShift.  You are right though, NightShift is kind of a Start Menu or hub for the rest of these applications.  The idea was to have an application-specific desktop where I could quickly pull up current conditions, read prior logs, track ToDo's and view my list of potential targets in anticipation of a night of imaging -- things I used to do "by hand".  All the heavy lifting was assigned to other apps which could be launched from NightShift:  target selection (ImagePlanner, sequencing (Humason), surveying (SuperScan, VariScan and ExoScan), satellite/asteroid imaging (HotPursuit), and a host of calibration tools that I wrote for TSX (autoguide, autofocus, PEC, etc).  

Frankly, I never really even tried to develop this idea and I never generalized it to configure for anything other than my own personal systems and locations -- you're the first to really tear into it, to my knowledge.   I figured that someone who was truly interested would either spawn their own version and take it from there (as you did), or there was sufficient interest that I would get off my ass and make it useful for someone other than myself.   So far the jury is still out -- and I'm OK with that.
